internal/proxy/hook_interceptor.go

```diff
@@ -17,38 +17,47 @@ import (
 
 var hoo hook.Hook
 
+type restfulHandler func(ctx context.Context, req any, userName, fullMethod string, handler grpc.UnaryHandler) (interface{}, error)
```

Comment:
rename ,do not include restful
This thing is being utilized not only for "restful" purposes.